spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DatastoreSimpleTypes.java

```diff
@@ -0,0 +1,132 @@
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastoreDataException;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastorePersistentProperty;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.data.mapping.model.SimpleTypeHolder;
+
+/**
+ * A class to manage Datastore-specific simple type conversions.
+ *
+ * @author Dmitry Solomakha
+ *
+ * @since 1.1
+ */
+class DatastoreSimpleTypes {
+	static final Map<Class<?>, Function<?, Value<?>>> DATASTORE_TYPE_WRAPPERS;
+
+	static final Set<Class<?>> DATASTORE_NATIVE_TYPES;
+
+	static final Set<Class<?>> ID_TYPES;
+
+	static {
+		//keys are used for type resolution, in order of insertion
+		DATASTORE_TYPE_WRAPPERS = ImmutableMap.<Class<?>, Function<?, Value<?>>>builder()
+				.put(Boolean.class, (Function<Boolean, Value<?>>) BooleanValue::of)
+				.put(Long.class, (Function<Long, Value<?>>) LongValue::of)
+				.put(Double.class, (Function<Double, Value<?>>) DoubleValue::of)
+				.put(LatLng.class, (Function<LatLng, Value<?>>) LatLngValue::of)
+				.put(Timestamp.class, (Function<Timestamp, Value<?>>) TimestampValue::of)
+				.put(String.class, (Function<String, Value<?>>) StringValue::of)
+				.put(Blob.class, (Function<Blob, Value<?>>) BlobValue::of)
+				.put(Entity.class, (Function<Entity, Value<?>>) EntityValue::of)
+				.put(Key.class, (Function<Key, Value<?>>) KeyValue::of)
+				.build();
+
+
+		//entries are used for type resolution, in order of insertion
+		DATASTORE_NATIVE_TYPES = ImmutableSet.<Class<?>>builder()
+				.addAll(DATASTORE_TYPE_WRAPPERS.keySet())
+				.build();
+
+		ID_TYPES = ImmutableSet.<Class<?>>builder()
+				.add(String.class)
+				.add(Long.class)
+				.build();
+	}
+
+	static final SimpleTypeHolder HOLDER = new SimpleTypeHolder(DATASTORE_NATIVE_TYPES, true);
+
+	final private Map<Class, Optional<Class<?>>> writeConverters = new HashMap<>();
+
+	final private ConversionService conversionService;
+
+	DatastoreSimpleTypes(ConversionService conversionService) {
+		this.conversionService = conversionService;
+	}
+
+	static boolean isSimple(Class aClass) {
+		return aClass == null || DATASTORE_NATIVE_TYPES.contains(aClass);
+	}
+
+	@SuppressWarnings("unchecked")
+	static public Value getDatastoreWrappedValue(Object propertyVal, DatastorePersistentProperty persistentProperty) {
+		if (propertyVal == null) {
+			return new NullValue();
+		}
+		Function wrapper = DatastoreSimpleTypes.DATASTORE_TYPE_WRAPPERS.get(propertyVal.getClass());
+		if (wrapper != null) {
+			return (Value) wrapper.apply(propertyVal);
+		}
+		throw new DatastoreDataException("Unable to convert a property" +
+				" with name " + persistentProperty.getFieldName()
+				+ " to Datastore supported type. The property's type is " + propertyVal.getClass());
+	}
+
+
+	public Optional<Class<?>> getCustomWriteTarget(Class<?> sourceType) {
```

Comment:
This one, and the method below are only used in `TwoStepsConversion`. Can we just move the two methods there?